Mastodon/Scene/Root/MainTab/MainTabBarController.swift

```diff
@@ -170,145 +171,151 @@ extension MainTabBarController {
         view.backgroundColor = .systemBackground
 
         // seealso: `ThemeService.apply(theme:)`
-        let tabs = Tab.allCases
-        let viewControllers: [UIViewController] = tabs.map { tab in
-            let viewController = tab.viewController(context: context, authContext: authContext, coordinator: coordinator)
-            viewController.tabBarItem.tag = tab.tag
-            viewController.tabBarItem.title = tab.title     // needs for acessiblity large content label
-            viewController.tabBarItem.image = tab.image.imageWithoutBaseline()
-            viewController.tabBarItem.largeContentSizeImage = tab.largeImage.imageWithoutBaseline()
-            viewController.tabBarItem.accessibilityLabel = tab.title
-            viewController.tabBarItem.accessibilityUserInputLabels = tab.inputLabels
-            viewController.tabBarItem.imageInsets = UIEdgeInsets(top: 6, left: 0, bottom: -6, right: 0)
-            return viewController
-        }
-        _viewControllers = viewControllers
-        setViewControllers(viewControllers, animated: false)
-        selectedIndex = 0
-
-        // hacky workaround for FB11986255 (Setting accessibilityUserInputLabels on a UITabBarItem has no effect)
-        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(50)) {
-            if let searchItem = self.tabBar.subviews.first(where: { $0.accessibilityLabel == Tab.search.title }) {
-                searchItem.accessibilityUserInputLabels = Tab.search.inputLabels
+        Task { @MainActor in
```

Comment:
Why is this (and the await in l. 179) needed?

`viewDidLoad` should run on `Main` nevertheless, why start a Task on `Main` to do `Main`-work? I'm sure: I'm missing something :)

Reply:
That's because the method in [L:179](https://github.com/mastodon/mastodon-ios/pull/1167/files/c1ad3e2a9c7cabe81f99920b1bfda396d45a2a82#diff-4f9c4dc8589cb44d4bbf40a2b9ed397b53261d8995625b1b8e760d35341650d8R179) is an `async` method which as there _was_ a time (briefly while refactoring) when the `ProfileViewModel` did some async work in it's contractor. That being said, this wasn't necessary anymore and I've removed it now.

Very very good catch 🦅 👁️ !